### PR TITLE
[[ Bug 15263 ]] Make sure file size uses off_t in Open (7.0).

### PR DESCRIPTION
**Pull request by runrevmark**

### CODE REVIEW

**Comment by peter-b** [comment]
@livecode-vulcan review ok ef6cc2a

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok ef6cc2a5686b6eac8572afc237af4211987185cf

**Comment by livecode-vulcan** [comment]
😞 test failure ef6cc2a5686b6eac8572afc237af4211987185cf
- try-linux-x86: [success](https://vulcan.livecode.com/builders/try-linux-x86/builds/328)
- try-win-x86: [pending](https://vulcan.livecode.com/builders/try-win-x86/builds/391)
- try-ios: [failure](https://vulcan.livecode.com/builders/try-ios/builds/273)
- try-android-armv6: [success](https://vulcan.livecode.com/builders/try-android-armv6/builds/339)
- try-linux-x86_64: [failure](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/274)
- try-mac: [failure](https://vulcan.livecode.com/builders/try-mac/builds/264)

**Comment by peter-b** [comment]
:octopus: :octopus: :octopus: 

@livecode-vulcan review ok f0f6b50

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok f0f6b50832054e46825468a326b0580cadcb3590

**Comment by livecode-vulcan** [comment]
😞 test failure f0f6b50832054e46825468a326b0580cadcb3590
- try-linux-x86: [pending](https://vulcan.livecode.com/builders/try-linux-x86/builds/329)
- try-win-x86: [pending](https://vulcan.livecode.com/builders/try-win-x86/builds/392)
- try-ios: [pending](https://vulcan.livecode.com/builders/try-ios/builds/274)
- try-android-armv6: [pending](https://vulcan.livecode.com/builders/try-android-armv6/builds/340)
- try-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/275)
- try-mac: [failure](https://vulcan.livecode.com/builders/try-mac/builds/265)

File: engine/src/dskmac.cpp [hunk 1]
```diff
@@ -6581,11 +6581,11 @@ struct MCMacDesktop: public MCSystemInterface, public MCMacSystemService
         // Using a memory mapped file now also possible for Mac
         if (p_map && MCmmap && p_mode == kMCOpenFileModeRead)
         {
-            int t_fd = open(*t_path_utf, O_RDONLY);
```
Comment (peter-b) [inline engine/src/dskmac.cpp:6584]: This change is incorrect!

**Comment by peter-b** [comment]
@livecode-vulcan review ok ff68883

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok ff688837989389f472d561ebb4d6780d8212449c

**Comment by livecode-vulcan** [comment]
😎 test success ff688837989389f472d561ebb4d6780d8212449c
- try-linux-x86: [success](https://vulcan.livecode.com/builders/try-linux-x86/builds/331)
- try-win-x86: [success](https://vulcan.livecode.com/builders/try-win-x86/builds/394)
- try-ios: [success](https://vulcan.livecode.com/builders/try-ios/builds/276)
- try-android-armv6: [success](https://vulcan.livecode.com/builders/try-android-armv6/builds/342)
- try-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/277)
- try-mac: [success](https://vulcan.livecode.com/builders/try-mac/builds/267)

**Pull request merged**